Proposal: v2_headers flag to inherit parentHeaders

appden opened the pull request:
I've found it very surprising and have been bitten a few times that a parent route's headers are not automatically used by a child route when that child route does not export its own `headers`. This proposes doing just that for v2 since this would be a breaking change.

MichaelDeBoey commented:
Hi @appden!

Are you still interested in getting this one merged?

If so, please rebase onto latest `dev` & fix failing tests

appden commented:
Thanks @MichaelDeBoey, I just rebased and fixed that type error for the tests. If this proposal sounds good (I recall @ryanflorence also wanted to take a look), I can add a test for this new functionality and also update to default this new flag to `true` on the templates.

appden commented:
@MichaelDeBoey @ryanflorence kindly following up on this, thanks!

brophdawg11 commented:
Thanks @appden!  It turns out this was never the intentional behavior, and prior to the v1 release it behaved the way you would expect (the way this Pr is changing it to behave).  We agree it's best to put this behind a v2 flag since it released in v1 with the current behavior.

This is going to change/conflict slightly with some work I just merged for `errorHeaders` in #6425 so I'm going to merge this into a branch of mine and get it synced up and then merge it to dev.  I'll keep your name on the commits throughout as a co-author!

appden commented:
Awesome, thanks @brophdawg11!

brophdawg11 commented:
Continued in https://github.com/remix-run/remix/pull/6431
